ansible/roles/openshift_template_deployer/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+---
+# Ensure needed variables are set
+# This is a a place to verify needed values
+- name: Ensure that osvp is set
+  fail:
+    msg:  "Please ensure that 'osvp_<variables>' are set!"
+  when: item is undefined
+  with_items:
+  - "{{ ostd_template_name }}"
+  - "{{ ostd_template_file }}"
+  - "{{ ostd_project }}"
+  run_once: true
+
+# install rpm if needed aos_3.5_roles/openshift_hosted_templates/
+- name: install the "{{ ostd_rpm }}"
+  yum:
+    name: "{{ ostd_rpm }}"
+    state: installed
+  when: ostd_rpm
+
+- name: Make template changes if needed
+  yedit:
+    src: "{{ ostd_template_file }}"
+    key: "{{ item.key }}"
+    value: "{{ item.value }}"
+  when: ostd_template_edits
+  with_items: "{{ ostd_template_edits }}"
```

Comment:
In addition, if the task protected by the `when:` uses the variable, I needed to do something like "{{ ostd_template_edits | default(omit) }}" or else I would get an error that the variable was undefined.